src/steth/scDAI.sol

```diff
@@ -353,6 +353,16 @@ contract scDAI is BaseV2Vault {
         return _calculateWethProfit(wethInvested(), totalDebt());
     }
 
+    /**
+     * @notice Swap Weth for sDAI
+     * @param _wethAmount Amount of weth to swap for sDAI
+     * @param _sDaiAmountOutMin Minimum amount of sDAI to remain after the swap
+     */
+    function swapWethForAsset(uint256 _wethAmount, uint256 _sDaiAmountOutMin) external returns (uint256) {
+        _onlyKeeper();
+        return _swapWethForAsset(_wethAmount, _sDaiAmountOutMin);
+    }
+
```

Comment:
move this function up so it is before all of the view functions to maintain consistency. and since it is a state-changing function, it should also emit an event, but be mindful to only emit the event inside the external function, not inside `_swapWethForAsset`. don't forget to also test for the event emission

Reply:
[Done](https://github.com/lindy-labs/sandclock-contracts/pull/155/commits/96d047a21afd11f676a0c029c7471f7648c19aed)